pkg/serviceaccounttoken/secret_test.go

```diff
@@ -193,3 +193,158 @@ func TestEnsureSecretForServiceAccount(t *testing.T) {
+			Labels: map[string]string{
+				ServiceAccountSecretLabel: baseSA.Name,
+			},
+			Annotations: map[string]string{
+				serviceAccountSecretAnnotation: baseSA.Name,
+			},
+		},
+		Type: v1.SecretTypeOpaque,
+	}
+	invalidSecretAnnotation := v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "invalid-secret-annotation",
+			Namespace: "test-ns",
+			Labels: map[string]string{
+				ServiceAccountSecretLabel: baseSA.Name,
+			},
+			Annotations: map[string]string{
+				serviceAccountSecretAnnotation: "some-other-sa",
+			},
+		},
+		Type: v1.SecretTypeOpaque,
+	}
+	tests := []struct {
+		name       string
+		stateSetup func(testState)
+		inputSA    *v1.ServiceAccount
+		wantSecret *v1.Secret
+		wantError  bool
+	}{
+		{
+			name:      "test nil sa",
+			inputSA:   nil,
+			wantError: true,
+		},
+		{
+			name:       "test no secrets",
+			inputSA:    &baseSA,
+			wantError:  false,
+			wantSecret: nil,
+		},
+		{
+			name:    "test valid secrets, first returned",
+			inputSA: &baseSA,
+			stateSetup: func(ts testState) {
+				validSecondSecret := validSecret.DeepCopy()
+				validSecondSecret.Name = "base-sa-secret-2"
+				ts.fakeLister.secrets = []*v1.Secret{&validSecret, validSecondSecret}
+			},
+			wantError:  false,
+			wantSecret: &validSecret,
+		},
+		{
+			name:    "test invalid secrets, none returned",
+			inputSA: &baseSA,
+			stateSetup: func(ts testState) {
+				ts.fakeLister.secrets = []*v1.Secret{&invalidSecretType, &invalidSecretAnnotation}
+				ts.clientset.Tracker().Add(&invalidSecretType)
+				ts.clientset.Tracker().Add(&invalidSecretAnnotation)
+			},
+			wantError:  false,
+			wantSecret: nil,
+		},
+		{
+			name:    "test invalid secrets delete failure, valid still returned",
+			inputSA: &baseSA,
+			stateSetup: func(ts testState) {
+				ts.fakeLister.secrets = []*v1.Secret{&invalidSecretType, &invalidSecretAnnotation, &validSecret}
+				ts.clientset.Tracker().Add(&invalidSecretType)
+				// don't add the invalid annotation secret to the state, this will cause a not-found error on delete
+			},
+			wantError:  false,
+			wantSecret: &validSecret,
+		},
+		{
+			name:    "test valid + invalid secrets, only valid returned",
+			inputSA: &baseSA,
+			stateSetup: func(ts testState) {
+				ts.fakeLister.secrets = []*v1.Secret{&invalidSecretType, &invalidSecretAnnotation, &validSecret}
+				ts.clientset.Tracker().Add(&invalidSecretType)
+				ts.clientset.Tracker().Add(&invalidSecretAnnotation)
+			},
+			wantError:  false,
+			wantSecret: &validSecret,
+		},
+		{
+			name:    "test secret lister error",
+			inputSA: &baseSA,
+			stateSetup: func(ts testState) {
+				ts.fakeLister.secrets = []*v1.Secret{&invalidSecretType, &invalidSecretAnnotation, &validSecret}
+				ts.fakeLister.err = fmt.Errorf("server unavailable")
+			},
+			wantError: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			k8sClient := fake.NewSimpleClientset()
+			k8sClient.CoreV1().Secrets("")
```

Comment:
What is the purpose of this line, to initialize the controller or anything?

Reply:
No purpose, just some left-over code from when I was getting the tests setup. Good catch, I've removed this.